clang-tools-extra/docs/clang-tidy/checks/cppcoreguidelines/pro-bounds-avoid-unchecked-container-accesses.rst

```diff
@@ -0,0 +1,51 @@
+.. title:: clang-tidy - cppcoreguidelines-pro-bounds-avoid-unchecked-container-accesses
+
+cppcoreguidelines-pro-bounds-avoid-unchecked-container-accesses
+===============================================================
+
+Flags the unsafe ``operator[]``. Can suggests fixing it with ``at()`` or a
+user-provided function.
+
+For example the code
+
+.. code-block:: c++
+
+  std::vector<int> a;
+  int b = a[4];
+
+will generate a warning but 
+
+.. code-block:: c++
+
+  std::unique_ptr<vector> a;
+  int b = a[0];
+
+will generate a warning.
+
+The classes ``std::map``, ``std::unordered_map`` and ``std::flat_map`` are
```

Comment:
It'd be good to not have to maintain this list both in this file and in the implementation. Can we make it more generic, like "STL containers with well-defined behavior for ``operator[]`` are excluded from this check".